test/e2e/12_sync.bats

```diff
@@ -45,7 +47,38 @@ function setup() {
   [ "$sync_tag_hash" = "$head_hash" ]
 }
 
+@test "Sync fails on duplicate resource" {
+  # Wait until flux deploys the workloads
+  poll_until_true 'workload podinfo' 'kubectl -n demo describe deployment/podinfo'
+
+  # Check the sync tag
+  git pull -f --tags
+  local sync_tag_hash
+  sync_tag_hash=$(git rev-list -n 1 flux)
+  head_hash=$(git rev-list -n 1 HEAD)
+  [ "$sync_tag_hash" = "$head_hash" ]
+
+  # Bump the image of podinfo, duplicate the resource definition (to cause a sync failure)
+  # and make sure the sync doesn't go through
+  sed -i'.bak' 's%stefanprodan/podinfo:2.1.0%stefanprodan/podinfo:3.1.5%' "${clone_dir}/workloads/podinfo-dep.yaml"
```

Comment:
```suggestion
  sed -i'.bak' 's%stefanprodan/podinfo:.*%stefanprodan/podinfo:3.1.5%' "${clone_dir}/workloads/podinfo-dep.yaml"
```

Will make it immune to upstream version changes (except if upstream changes it the tag to `3.1.5`).

Reply:
(When changing this you probably also want to change the replace in the basic sync test).